extras/systemd/dynflow-sidekiq@.service

```diff
@@ -4,13 +4,23 @@ Documentation=https://theforeman.org
 After=network.target remote-fs.target nss-lookup.target
 
 [Service]
-Type=forking
+Type=simple
 User=foreman
 TimeoutSec=90
 Environment=RAILS_ENV=production
 Environment=DYNFLOW_SIDEKIQ_SCRIPT=/usr/share/foreman/extras/dynflow-sidekiq.rb
+# Greatly reduce Ruby memory fragmentation and heap usage
+# https://www.mikeperham.com/2018/04/25/taming-rails-memory-bloat/
+Environment=MALLOC_ARENA_MAX=2
 WorkingDirectory=/usr/share/foreman
 ExecStart=/usr/bin/sidekiq -e ${RAILS_ENV} -r ${DYNFLOW_SIDEKIQ_SCRIPT} -C /etc/foreman/dynflow/%i.yml
+ExecReload=/usr/bin/kill -TSTP $MAINPID
```

Comment:
Educational question -- what is this doing?

Reply:
See - https://github.com/mperham/sidekiq/wiki/Signals it is traped in sidekiq not to shutdown, only to stop processing the jobs. So you have you can send it reload to inform it, that it's going to get restarted soon.